src/pages/add-lock-exception/AddLockExceptionForm/OrganisationUnitSelection/OrganisationUnitSelection.js

```diff
@@ -0,0 +1,154 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import i18n from '@dhis2/d2-i18n'
+import {
+    NoticeBox,
+    CenteredContent,
+    CircularLoader,
+    OrganisationUnitTree,
+} from '@dhis2/ui'
+import PropTypes from 'prop-types'
+import React, { useState, useEffect } from 'react'
+import styles from './OrganisationUnitSelection.module.css'
+import SelectAll from './SelectAll'
+import SelectByGroup from './SelectByGroup'
+import SelectByLevel from './SelectByLevel'
+
+const union = (arr1, arr2) => [...new Set(arr1.concat(arr2))]
```

Comment:
`union` and `difference` are set operations used by `handleSelect` and `handleDeselect` respectively (see below).

Set operations were used for updating the org unit selections for performance reasons: the O(n^2) methods used by the `d2-ui` OrgUnitSelect components (see [here](https://github.com/dhis2/d2-ui/blob/d0d3812fe70b8b6f6c33cf0f85d89ba987f8b8c1/packages/org-unit-select/src/common.js#L23-L24) and [here](https://github.com/dhis2/d2-ui/blob/d0d3812fe70b8b6f6c33cf0f85d89ba987f8b8c1/packages/org-unit-select/src/common.js#L31-L34)) took ~100ms on Sierra Leone db instances, while the set operations took ~1ms.